File: lab 3/RecurringEvent.cpp
#include <iostream>
#include "RecurringEvent.h"
using namespace std;

RecurringEvent::RecurringEvent() :Event()
{
    for (int& i : startDate) {
        i = 0;
    }
    for (int& i : endDate) {
        i = 0;
    }
    repeats = 0;
}

RecurringEvent::RecurringEvent(const string& name, double duration, int sd, int sm, int sy,
    int ed, int em, int ey, int repeats)
    : Event(name, duration), repeats(repeats)
{
    startDate[0] = sd; startDate[1] = sm; startDate[2] = sy;
    endDate[0] = ed; endDate[1] = em; endDate[2] = ey;

}

void RecurringEvent::displayEvent() const
{
    cout << "Recurring Event: " << name 
        << ",  " << startDate[0] << "." << startDate[1] << "." << startDate[2]
        << " - " << endDate[0] << "." << endDate[1] << "." << endDate[2]
         << ", Repeats: " << repeats << ", Duration: " << duration << endl;
}

int RecurringEvent::getRepeats() const { return repeats; }
const int* RecurringEvent::getStartDate() const { return startDate;}
const int* RecurringEvent::getEndDate() const {return endDate;}


RecurringEvent& RecurringEvent::inputEventDetails()
{
    cout << "Enter information about event: name, duration, start date(day, month, year),  end date(day, month, year), repeats" << endl;
    cin >> name >> duration >> startDate[0] >> startDate[1] >> startDate[2] >> endDate[0] >> endDate[1] >> endDate[2] >> repeats;
    return *this;
}

File: python/RecurringEvent.py
from Event import Event

class ReccuringEvent(Event):
    def __init__(self, line, dur, sdd, smm, syy, edd, emm, eyy, rep):
        super().__init__(line, dur)
        self._start_date=[sdd, smm, syy]
        self._end_date=[edd, emm, eyy]
        self._repeats=rep
    
    def get_start_date(self):
        return self._start_date
        
    def get_end_date(self):
        return self._end_date
    
    def display_event(self):
        print(f"{self._name}: {self._start_date[0]}.{self._start_date[1]}.{self._start_date[2]}-{self._end_date[0]}.{self._end_date[1]}.{self._end_date[2]}, Repeats: {self._repeats}, Duration: {self._duration}")
        
    def input_event_details(self):
        print(f"Enter information about event: ")
        self._name=input("Name: ")
        self._duration=float(input("Duration: "))
        print(f"Start date:")
        self._start_date[0]=int(input("day: "))
        self._start_date[1]=int(input("month: "))
        self._start_date[2]=int(input("year: "))
        print(f"End date:")
        self._end_date[0]=int(input("day: "))
        self._end_date[1]=int(input("month: "))
        self._end_date[2]=int(input("year: "))
        self._repeats=int(input("repeats: "))
